src/webgldrawer.js

```diff
@@ -206,6 +207,25 @@
             return canvas;
         }
 
+        /**
+         * Get the backup renderer (CanvasDrawer) to use if data cannot be used by webgl
+         * Lazy loaded
+         * @private
+         * @returns {CanvasDrawer}
+         */
+        _getBackupCanvasDrawer(){
+            if(!this._backupCanvasDrawer){
+                this._backupCanvasDrawer = new $.CanvasDrawer({
+                    viewer: this.viewer,
+                    viewport: this.viewport,
+                    element: this.container,
```

Comment:
`debugGridColor` is the only "missing" option as far as I can tell, and it is optional. I'm not sure whether passing through the `WebGLDrawer`'s value for this option would make sense or not.

My goal is to provide a reasonable fallback drawing implementation, but if a default debug color is used instead of the dev-specified one I hope they will just see the warning message in the console and fix their CORS settings first.

Reply:
I was having additional parameters in the cache refactoring shared by all drawers, that's why I mention it here.

Reply:
@Aiosa What would you propose here to make sure the additional parameters are passed through?

Reply:
to provide the target options from the `drawerOptions` map... with new cache system a drawer can e.g. specify whether to store its data as internal cache, or re-use the 'main' tile cache, which can be toggled by users exactly through these options, and all dravers should get them

Reply:
Cool. What do you think @pearcetm?

Reply:
I think it's a good idea. I'm planning to consolidate the drawer creation code into an internal private API on the viewer so that the same initialization and options are always used. I'm also planning to add a `drawer-error` event (I'm open to suggestions on the naming of this event) that can be fired when something goes wrong with drawing a particular TiledImage. After that I'll fix up whatever merge conflicts there are. It may take me a week or more to get to these things.

Reply:
@pearcetm Sounds good!